worker/src/response.rs

```diff
@@ -356,9 +347,73 @@ fn no_using_invalid_error_status_code() {
 }
 
 pub struct ResponseInit {
-    pub status: u16,
-    pub headers: Headers,
-    pub websocket: Option<WebSocket>,
+    status: u16,
+    headers: Headers,
+    websocket: Option<WebSocket>,
+    encode_body: Option<String>,
+    cf: Option<serde_json::Value>,
+}
+
+impl ResponseInit {
+    pub fn builder() -> ResponseInitBuilder {
+        ResponseInitBuilder::new()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ResponseInitBuilder {
+    status_code: u16,
+    headers: Headers,
+    websocket: Option<WebSocket>,
```

Comment:
Is it actually valid to clone a `WebSocket`. In https://github.com/cloudflare/workers-rs/commit/9042648b8a5ce6879a1f36ff9b278ad7258d7c15, I see `WebSocket` explicitly cloned, but I also see an explicit check to prevent cloning `WebSocket` in the same commit.

Reply:
It is valid to clone the `WebSocket` struct since you're just cloning a *reference* to a JavaScript WebSocket object, but when we call `.clone()` on `EdgeResponse` we're actually calling [a JS clone method](https://developer.mozilla.org/en-US/docs/Web/API/Response/clone). In workerd this'll throw if the `ResponseInit` contains a websocket, so we check before hand to ensure this doesn't throw an Error in JS.